tritium-metrics/src/test/java/com/palantir/tritium/metrics/InstrumentedSslContextTest.java

```diff
@@ -176,6 +177,23 @@ public void testServerInstrumentation() throws Exception {
         assertThat(metrics.meter(name).getCount()).isEqualTo(1);
     }
 
+    @Test
+    public void testSslEngineUnwrapNotInstrumented() throws IOException, GeneralSecurityException {
+        SSLEngine engine = newServerContext().createSSLEngine();
+        assertThat(MetricRegistries.unwrap(engine)).isSameAs(engine);
+    }
+
+    @Test
+    public void testSslEngineUnwrapInstrumentedTwice() throws IOException, GeneralSecurityException {
+        SSLContext context = newServerContext();
+        TaggedMetricRegistry metrics = new DefaultTaggedMetricRegistry();
+        SSLContext wrapped = MetricRegistries.instrument(
+                metrics, MetricRegistries.instrument(metrics, context, "first"), "second");
+        SSLEngine engine = wrapped.createSSLEngine();
+        assertThat(engine).isInstanceOf(InstrumentedSslEngine.class);
+        assertThat(MetricRegistries.unwrap(engine)).isNotInstanceOf(InstrumentedSslEngine.class);
```

Comment:
Probably worth adding `.isNotNull()` as well just to be sure (though that should be impossible)

```suggestion
        assertThat(MetricRegistries.unwrap(engine)).isNotNull().isNotInstanceOf(InstrumentedSslEngine.class);
```

Reply:
Good call. Updated `MetricRegistries.unwrap(SSLEngine)` with a null check as well.